src/test/regress/expected/alter_table_add_column.out

```diff
@@ -44,6 +44,15 @@ ERROR:  cannot execute ADD COLUMN command with PRIMARY KEY, UNIQUE, FOREIGN and
 DETAIL:  Adding a column with a constraint in one command is not supported because all constraints in Citus must have explicit names
 HINT:  You can issue each command separately such as ALTER TABLE referencing ADD COLUMN test_8 data_type; ALTER TABLE referencing ADD CONSTRAINT constraint_name CHECK (check_expression);
 ALTER TABLE referencing ADD COLUMN test_8 integer CONSTRAINT check_test_8 CHECK (test_8 > 0);
+-- error out properly even if the REFERENCES does not include the column list of the referenced table
```

Comment:
Having the PR description include that this is specific the case we're fixing would be good.